src/extension.ts

```diff
@@ -1067,6 +1065,34 @@ const getActivePythonFile = () => {
     return activeEditor ? activeEditor.document.fileName : "";
 };
 
+const updateCurrentFileFromEditor = async (
+    activeTextDocument: vscode.TextEditor | undefined,
+    currentPanel: vscode.WebviewPanel
+) => {
+    if (
+        activeTextDocument &&
+        activeTextDocument.document &&
+        activeTextDocument.document.languageId === "python"
+    ) {
+        setPathAndSendMessage(
+            currentPanel,
+            activeTextDocument.document.fileName
+        );
+        currentTextDocument = activeTextDocument.document;
+    } else if (currentFileAbsPath === "") {
+        setPathAndSendMessage(currentPanel, getActivePythonFile() || "");
+    }
+    if (
+        currentTextDocument &&
+        utils.getActiveEditorFromPath(currentTextDocument.fileName) ===
+            undefined
+    ) {
+        await vscode.window.showTextDocument(
+            currentTextDocument,
+            vscode.ViewColumn.One
+        );
+    }
+};
 const updateCurrentFileIfPython = async (
```

Comment:
Is this still relevant with `updateCurrentFileFromEditor` added? 

It seems like the only difference is whether we access `activeTextDocument.document`. If that's the case, you can add an extra parameter to verify from the document field (e.g. a boolean parameter, or either pass activeTextDocument.document or activeTextDocument as a parameter).

Reply:
Will do! I am currently working on a refactoring for file selection :)